Error when loading data when using OpenJDK 11

When switching to OpenJDK 11, I get errors in loading data. Works with OpenJDK 8.
I took care to compile and run with the same version.
(Did not recompile webknossos-wrap though, not sure how this is handled. Using wkw 1.1.7, built 2019-02-04)

```
backend: ! @7el0770p0 - Internal server error, for (POST) [/data/datasets/sample_organization/<datasetname>/layers/color/data?token=<token>] ->
backend:  
backend: play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[IllegalArgumentException: Cannot locate field unsafe on class java.nio.DirectByteBufferR]]
backend: 	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:351)
backend: 	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:267)
backend: 	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:448)
backend: 	at play.core.server.AkkaHttpServer$$anonfun$1.applyOrElse(AkkaHttpServer.scala:446)
backend: 	at scala.concurrent.Future.$anonfun$recoverWith$1(Future.scala:413)
backend: 	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:37)
backend: 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
backend: 	at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
backend: 	at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:92)
backend: 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
backend: Caused by: java.lang.IllegalArgumentException: Cannot locate field unsafe on class java.nio.DirectByteBufferR
backend: 	at org.apache.commons.lang3.Validate.isTrue(Validate.java:158)
backend: 	at org.apache.commons.lang3.reflect.FieldUtils.readField(FieldUtils.java:467)
backend: 	at com.scalableminds.webknossos.wrap.util.ExtendedTypes$ExtendedMappedByteBuffer.<init>(ExtendedTypes.scala:33)
backend: 	at com.scalableminds.webknossos.wrap.WKWFile.$anonfun$mapBuffers$1(WKWFile.scala:126)
backend: 	at com.scalableminds.webknossos.wrap.WKWFile.$anonfun$mapBuffers$1$adapted(WKWFile.scala:124)
backend: 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:233)
backend: 	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:32)
backend: 	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:29)
backend: 	at scala.collection.mutable.ArrayOps$ofLong.foreach(ArrayOps.scala:254)
backend: 	at scala.collection.TraversableLike.map(TraversableLike.scala:233)
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -85,6 +85,10 @@ jobs:
           name: Build webknossos-tracingstore (sbt)
           command: docker-compose run base sbt -no-colors "project webknossosTracingstore" copyMessages compile stage
 
+      - run:
+          name: Assemble webknossos-datastore fat-jar (sbt)
+          command: docker-compose run base sbt -no-colors "project webknossosDatastore" assembly
+
       - save_cache:
           name: Save target cache
           key: target-cache-{{ checksum ".circleci/cache_version" }}-{{ .Branch }}-{{ .Revision }}

--- a/CHANGELOG.unreleased.md
+++ b/CHANGELOG.unreleased.md
@@ -12,6 +12,7 @@ For upgrade instructions, please check the [migration guide](MIGRATIONS.released
 
 ### Added
 - Added the possibility for admins to set long-running jobs to a “manually repaired” state. [#5530](https://github.com/scalableminds/webknossos/pull/5530)
+- Added compatibility with newer JREs, tested with 8, 11 and 14. [#5558](https://github.com/scalableminds/webknossos/pull/5558)
 
 ### Changed
 - "Center new Nodes" option was renamed to "Auto-center Nodes" and changed to also influence the centering-behavior when deleting a node. [#5538](https://github.com/scalableminds/webknossos/pull/5538)

--- a/build.sbt
+++ b/build.sbt
@@ -68,20 +68,6 @@ lazy val webknossosDatastore = (project in file("webknossos-datastore"))
   .settings(
     name := "webknossos-datastore",
     commonSettings,
-    assembly / assemblyMergeStrategy := {
-      case PathList(ps @ _*) if ps.last endsWith ".class"                 => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last endsWith ".proto"                 => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last == "reference-overrides.conf"     => MergeStrategy.concat
-      case PathList(ps @ _*) if ps.last == "io.netty.versions.properties" => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last == "mailcap.default"              => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last == "mimetypes.default"            => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last == "native-image.properties"      => MergeStrategy.last
-      case PathList(ps @ _*) if ps.last == "application.conf"             => MergeStrategy.concat
-      case x =>
-        val oldStrategy = (assembly / assemblyMergeStrategy).value
-        oldStrategy(x)
-    },
-    assembly / test := {},
     BuildInfoSettings.webknossosDatastoreBuildInfoSettings,
     libraryDependencies ++= Dependencies.webknossosDatastoreDependencies,
     routesGenerator := InjectedRoutesGenerator,
@@ -94,7 +80,19 @@ lazy val webknossosDatastore = (project in file("webknossos-datastore"))
       }
       ((libs +++ subs +++ targets) ** "*.jar").classpath
     },
-    copyConfFilesSetting
+    copyConfFilesSetting,
+    assembly / assemblyMergeStrategy := {
+      case PathList(ps @ _*) if ps.last endsWith ".class" => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last endsWith ".proto" => MergeStrategy.last
+      case PathList(ps @ _*) if List("io.netty.versions.properties", "mailcap.default",
+        "mimetypes.default", "native-image.properties").contains(ps.last) => MergeStrategy.last
+      case PathList(ps @ _*) if List("application.conf", "reference-overrides.conf").contains(ps.last) => MergeStrategy.concat
+      case x =>
+        val oldStrategy = (assembly / assemblyMergeStrategy).value
+        oldStrategy(x)
+    },
+    assembly / test := {},
+    assembly / assemblyJarName := s"webknossos-datastore-${BuildInfoSettings.webKnossosVersion}.jar"
   )
 
 lazy val webknossosTracingstore = (project in file("webknossos-tracingstore"))

--- a/build.sbt
+++ b/build.sbt
@@ -25,6 +25,8 @@ ThisBuild / libraryDependencies ++= Seq(
   "com.github.ghik" % "silencer-lib" % "1.7.0" % Provided cross CrossVersion.full
 )
 
+ThisBuild / assemblyMergeStrategy := (_ => MergeStrategy.first)
+
 PlayKeys.devSettings := Seq("play.server.akka.requestTimeout" -> "10000s", "play.server.http.idleTimeout" -> "10000s")
 
 scapegoatIgnoredFiles := Seq(".*/Tables.scala",
@@ -68,6 +70,7 @@ lazy val webknossosDatastore = (project in file("webknossos-datastore"))
   .settings(
     name := "webknossos-datastore",
     commonSettings,
+    assembly / assemblyMergeStrategy := (_ => MergeStrategy.first),
     BuildInfoSettings.webknossosDatastoreBuildInfoSettings,
     libraryDependencies ++= Dependencies.webknossosDatastoreDependencies,
     routesGenerator := InjectedRoutesGenerator,

--- a/build.sbt
+++ b/build.sbt
@@ -25,8 +25,6 @@ ThisBuild / libraryDependencies ++= Seq(
   "com.github.ghik" % "silencer-lib" % "1.7.0" % Provided cross CrossVersion.full
 )
 
-ThisBuild / assemblyMergeStrategy := (_ => MergeStrategy.first)
-
 PlayKeys.devSettings := Seq("play.server.akka.requestTimeout" -> "10000s", "play.server.http.idleTimeout" -> "10000s")
 
 scapegoatIgnoredFiles := Seq(".*/Tables.scala",
@@ -70,7 +68,20 @@ lazy val webknossosDatastore = (project in file("webknossos-datastore"))
   .settings(
     name := "webknossos-datastore",
     commonSettings,
-    assembly / assemblyMergeStrategy := (_ => MergeStrategy.first),
+    assembly / assemblyMergeStrategy := {
+      case PathList(ps @ _*) if ps.last endsWith ".class"                 => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last endsWith ".proto"                 => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last == "reference-overrides.conf"     => MergeStrategy.concat
+      case PathList(ps @ _*) if ps.last == "io.netty.versions.properties" => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last == "mailcap.default"              => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last == "mimetypes.default"            => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last == "native-image.properties"      => MergeStrategy.last
+      case PathList(ps @ _*) if ps.last == "application.conf"             => MergeStrategy.concat
+      case x =>
+        val oldStrategy = (assembly / assemblyMergeStrategy).value
+        oldStrategy(x)
+    },
+    assembly / test := {},
     BuildInfoSettings.webknossosDatastoreBuildInfoSettings,
     libraryDependencies ++= Dependencies.webknossosDatastoreDependencies,
     routesGenerator := InjectedRoutesGenerator,

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -4,7 +4,7 @@ import sbt._
 object Dependencies {
   private val akkaVersion = "2.6.14"
   private val log4jVersion = "2.13.3"
-  private val webknossosWrapVersion = "1.1.11-SNAPSHOT"
+  private val webknossosWrapVersion = "1.1.11"
 
   private val akkaLogging = "com.typesafe.akka" %% "akka-slf4j" % akkaVersion
   private val akkaTest = "com.typesafe.akka" %% "akka-testkit" % akkaVersion

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -4,7 +4,7 @@ import sbt._
 object Dependencies {
   private val akkaVersion = "2.6.14"
   private val log4jVersion = "2.13.3"
-  private val webknossosWrapVersion = "1.1.10"
+  private val webknossosWrapVersion = "1.1.11-SNAPSHOT"
 
   private val akkaLogging = "com.typesafe.akka" %% "akka-slf4j" % akkaVersion
   private val akkaTest = "com.typesafe.akka" %% "akka-testkit" % akkaVersion

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -12,4 +12,6 @@ addSbtPlugin("com.sksamuel.scapegoat" %% "sbt-scapegoat" % "1.0.9")
 
 addSbtPlugin("net.vonbuchholtz" % "sbt-dependency-check" % "3.1.3")
 
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.15.0")
+
 libraryDependencies += "com.thesamet.scalapb" %% "compilerplugin" % "0.11.3"

